### PR TITLE
migrating flash to nf-test

### PR DESCRIPTION
**Pull request by DavideBag**
## PR checklist

- [x] Migrating flash module to nf-test.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/flash/main.nf [hunk 1]
```diff
@@ -19,10 +19,12 @@ process FLASH {
 
     script:
     def args = task.ext.args ?: ''
+    def args2 = task.ext.args ?: ''
```
Comment (SPPearce) [inline modules/nf-core/flash/main.nf:22]: What is this for?

**Review by SPPearce — COMMENTED** [review]
I don't understand what the args2 is for here, when it is just next to args in the code

File: modules/nf-core/flash/main.nf (hunk 1 above)
Comment (DavideBag) [reply modules/nf-core/flash/main.nf:22]: I inserted args2 to later add a --quiet function in the config file that I use in the pipeline.

Comment (DavideBag) [reply modules/nf-core/flash/main.nf:22]: Maybe I can put everything in args1?

Comment (SPPearce) [reply modules/nf-core/flash/main.nf:22]: Yes. args2 is used when multiple tools are being piped together (like bwa into samtools) so you can pass separate arguments to each tool ,or when tools have positional arguments (which isn't very common)

Comment (DavideBag) [reply modules/nf-core/flash/main.nf:22]: Thank you for the review and for the clarity, I'll make the changes.

**Review by SPPearce — COMMENTED** [review]
Does this module take fastq files in and produce different ones? If so, can you include a check that the output files have different names

**Comment by SPPearce** [comment]
> Does this module take fastq files in and produce different ones? If so, can you include a check that the output files have different names

Or you could give them a suffix to make it clear they have been through this tool.

**Comment by lescai** [comment]
Hi @SPPearce and @DavideBag 
I understand that preference is always not to *hard-code* any prefix / suffix, and leave this to the config.
**however**, as Simon highlights one should include an explicit check whenever there's the risk of an overlap in input/output names. 
Best practice I'd suggest is the one adopted in the `samtools` module, since in sorting / slicing this is very common:
https://github.com/nf-core/modules/blob/master/modules/nf-core/samtools/sort/main.nf

**Comment by SPPearce** [comment]
There are definitely many fastq tools that don't follow that convention though

**Comment by SPPearce** [comment]
I'm happy to introduce breaking changes that make sense. One option is to
rename the output channel, then it will definitely fail and force people to
do something

**Review by lescai — APPROVED** [review]
LGTM now, quite thorough and now more explicit with flash documentation.

**Review by SPPearce — APPROVED** [review]

**Comment by DavideBag** [comment]
Thank you to @SPPearce and @lescai for investing time in reviewing.

**Pull request merged**